airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py

```diff
@@ -0,0 +1,188 @@
+    FieldPath,
+    HttpResponseBuilder,
+    RecordBuilder,
+    create_record_builder,
+    create_response_builder,
+    find_template,
+)
+from airbyte_protocol.models import SyncMode
+
+from .config import ACCOUNT_ID, ConfigBuilder
+from .request_builder import get_account_request, get_ad_sets_request, get_ads_request, get_campaigns_request
+from .response_builder import get_account_response
+from .utils import config, read_output
+
+
+def _stream_record(stream: str, cursor_field: str = "updated_time") -> RecordBuilder:
+    return create_record_builder(
+        response_template=find_template(stream, __file__),
+        records_path=FieldPath("data"),
+        record_cursor_path=FieldPath(cursor_field),
+    )
+
+
+def _stream_response(stream: str) -> HttpResponseBuilder:
+    return create_response_builder(
+        response_template=find_template(stream, __file__),
+        records_path=FieldPath("data"),
+    )
+
+
+class TestIncludeDeleted(TestCase):
+    account_id = ACCOUNT_ID
+    filter_statuses_flag = "filter_statuses"
+    statuses = ["ACTIVE", "ARCHIVED"]
+
+    @staticmethod
+    def _read(config_: ConfigBuilder, stream_name: str, expecting_exception: bool = False) -> EntrypointOutput:
+        return read_output(
+            config_builder=config_,
+            stream_name=stream_name,
+            sync_mode=SyncMode.incremental,
+            expecting_exception=expecting_exception,
+        )
+
+    @HttpMocker()
+    def test_ads_stream(self, http_mocker: HttpMocker):
+        http_mocker.get(
+            get_account_request().build(),
+            get_account_response(),
+        )
+        filters = (
+            '[{"field":"ad.effective_status","operator":"IN","value":["ACTIVE","ARCHIVED"]},'
+            '{"field":"ad.updated_time","operator":"GREATER_THAN","value":1672531200}]'
+        )
+        fields = [
+            "bid_type",
+            "account_id",
+            "campaign_id",
+            "adset_id",
+            "adlabels",
+            "bid_amount",
+            "bid_info",
+            "status",
+            "creative",
+            "id",
+            "updated_time",
+            "created_time",
+            "name",
+            "targeting",
+            "effective_status",
+            "last_updated_by_app_id",
+            "recommendations",
+            "source_ad_id",
+            "tracking_specs",
+            "conversion_specs",
+        ]
+
+        http_mocker.get(
+            get_ads_request().with_limit(100).with_filtering(filters).with_fields(fields).with_summary().build(),
+            _stream_response("ads").with_record(_stream_record("ads")).build(),
+        )
+
+        output = self._read(config().with_ad_statuses(self.statuses), "ads")
+        assert len(output.records) == 1
+        account_state = output.most_recent_state.dict()["stream_state"][self.account_id]
+        assert self.filter_statuses_flag in account_state, f"State should include `filter_statuses` flag"
```

Comment:
The error message is clear but without context about the "why", it is hard to understand why this is asserted here. Can we add more info on that?

Reply:
updated error message